### PR TITLE
Update antd and styled components

### PR DESCRIPTION
**Pull request by rymaju**
## Why

We arent doing routing, antd styling, or styled components!

## This PR

Adding craco antd, theming, and proper examples of styled components (replacing all instances of .less with equivalent styled component code). Also fixed routing issues where pages would be reloaded or history not pushed when navigating. Also added example navbar.

## Verification Steps

Used npm run start and interacted with the site to ensure css was working as before and routing was not reloading the pages and was saving history to the browser.

### CODE REVIEW

File: antd.customize.less [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@link-color: #1890ff;
+@primary-color: #1da57a;
+@font-family: 'IBM Plex Sans', sans-serif;
```
Comment (jackblanc) [inline antd.customize.less]: ⛏️ Newline at end of file

File: antd.customize.less [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@link-color: #1890ff;
```
Comment (jackblanc) [inline antd.customize.less]: We should define the colors with descriptive names as their own variables, and set `@link-color: @our-variable-color`

File: craco.config.js [hunk 3]
```diff
@@ -1,18 +1,19 @@
-const CracoLessPlugin = require('craco-less');
-
+const CracoAntDesignPlugin = require('craco-antd');
+process.env.BROWSER = "none";
```
Comment (jackblanc) [inline craco.config.js:2]: ❓ Why is this required?

File: src/App.tsx [hunk 4]
```diff
@@ -3,7 +3,6 @@ import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 
 // Import antd stylesheets
```
Comment (jackblanc) [inline src/App.tsx]: ⛏️ Let's remove this comment

File: src/colors.tsx [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const PRIMARY = '#1da57a';
```
Comment (jackblanc) [inline src/colors.tsx:1]: Same comment as less file, lets define the colors separately and assign values here to prevent confusion

File: src/components/index.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (jackblanc) [inline src/components/index.tsx]: What's this file for? Common utility components?

File: src/components/navbar/NavBar.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      <Menu.Item>Change Password</Menu.Item>
+      <Menu.Item>Deactivate Account</Menu.Item>
+    </Menu>
+  );
+
+  return (
+    <NavRow align="middle">
+      <Col flex={1}>
+        <Link to="/">
+          <Logo
```
Comment (jackblanc) [inline src/components/navbar/NavBar.tsx]: Let's remove the logo from the frontend-scaffold

File: src/containers/template-24-col-grid/Template.tsx [hunk 8]
```diff
@@ -10,6 +10,16 @@ AntD Components:
 https://ant.design/components/overview/
 */
 
+const GridContentContainer = styled.div`
+  padding: 24px;
+`;
+
+const ExampleCol = styled(Col)`
+  border: 1px #1890ff solid;
```
Comment (jackblanc) [inline src/containers/template-24-col-grid/Template.tsx:18]: Can we access our variables here?

**Review by jackblanc — APPROVED** [review]
Looks great! Left a few suggestions

File: src/components/index.tsx (hunk 6 above)
Comment (bensenescu) [reply src/components/index.tsx]: Yea, I'd probably just make a LinkButton component and then have a styles folder serperate from components for styles like ContentContainer.

File: src/components/navbar/NavBar.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  padding: 1em 0;
+`;
+const AntButtonLink = styled(LinkButton)`
+  color: black;
+  margin-left: 7em;
+`;
+const Logo = styled.img`
+  float: right;
+`;
+const LogoText = styled(Text)`
```
Comment (bensenescu) [inline src/components/navbar/NavBar.tsx]: Probably isn't necessary for the scaffold, but I would move all typography to that styles folder and always use your typography styled components for any text. Keeps the site more consistent.

File: src/components/navbar/NavBar.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  font-size: 24px;
+`;
+const UserDropdown = styled(Button)`
+  float: right;
+  margin-right: 3em;
```
Comment (bensenescu) [inline src/components/navbar/NavBar.tsx:18]: When I see this margin-right: 3em in two places next two each other, it makes me think "should this be a constant?"

File: src/components/navbar/NavBar.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+  return (
+    <NavRow align="middle">
+      <Col flex={1}>
+        <Link to="/">
+          <Logo
+            src={
+              'https://lucys-love-bus-public.s3.us-east-2.amazonaws.com/LLB_2019_Sq_rgb+1.png'
+            }
+            alt={'Site Logo'}
```
Comment (bensenescu) [inline src/components/navbar/NavBar.tsx]: alt="Site Logo" is how I prefer using strings.

File: src/components/navbar/NavBar.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+      </Col>
+
+      <Col flex={3}>
+        <AntButtonLink to="/">Home</AntButtonLink>
+        <AntButtonLink to="/block-template">Block Template</AntButtonLink>
+        <AntButtonLink to="/grid-template">Grid Template</AntButtonLink>
+      </Col>
+
+      <div>
+        {authenticated ? (
```
Comment (bensenescu) [inline src/components/navbar/NavBar.tsx:51]:
```suggestion
        {
        authenticated 
        ? (
```

File: src/components/navbar/NavBar.tsx [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+          <Col flex={3}>
+            <Dropdown overlay={userMenu}>
+              <UserDropdown>
+                <UserOutlined /> John Smith <DownOutlined />
+              </UserDropdown>
+            </Dropdown>
+          </Col>
+        ) : (
```
Comment (bensenescu) [inline src/components/navbar/NavBar.tsx:59]:
```suggestion
        ) 
        : (
```

File: src/containers/template-24-col-grid/Template.tsx (hunk 8 above)
Comment (bensenescu) [reply src/containers/template-24-col-grid/Template.tsx:18]: You'd have to pull them into a ts file, not a less file.

Comment (bensenescu) [reply src/containers/template-24-col-grid/Template.tsx:18]: But, you should be using constants.

**Review by bensenescu — APPROVED** [review]

File: craco.config.js (hunk 3 above)
Comment (rymaju) [reply craco.config.js:2]: This is required so nodemon doesnt constantly open a new tab in the browser every time it detects a change

File: src/colors.tsx (hunk 5 above)
Comment (rymaju) [reply src/colors.tsx:1]: Im a bit confused by what you mean by this? Can you elaborate and suggest what I should change? I dont think its possible to share one common file that will sync across both the antd less styles and colors as JS values.

File: src/components/index.tsx (hunk 6 above)
Comment (rymaju) [reply src/components/index.tsx]: I can move LinkButton out into its own component folder

File: src/components/navbar/NavBar.tsx (hunk 7 above)
Comment (rymaju) [reply src/components/navbar/NavBar.tsx]: Ok, this will break a bunch of CSS though

File: src/components/navbar/NavBar.tsx (hunk 9 above)
Comment (rymaju) [reply src/components/navbar/NavBar.tsx]: I might just remove most of the styling @anirudh723 did for the navbar for the frontend scaffold then
